pkg/gofr/datasource/kv-store/badger/badger.go

```diff
@@ -0,0 +1,125 @@
+package badger
+
+import (
+	"context"
+	"errors"
+	"github.com/dgraph-io/badger/v4"
+)
+
+type Configs struct {
+	DirPath string
+}
+
+type client struct {
+	db      *badger.DB
+	configs Configs
+	logger  Logger
+	metrics Metrics
+}
+
+func New(configs Configs) *client {
+	return &client{configs: configs}
+}
+
+// UseLogger sets the logger for the MongoDB client which asserts the Logger interface.
+func (c *client) UseLogger(logger any) {
+	if l, ok := logger.(Logger); ok {
+		c.logger = l
+	}
+}
+
+// UseMetrics sets the metrics for the MongoDB client which asserts the Metrics interface.
+func (c *client) UseMetrics(metrics any) {
+	if m, ok := metrics.(Metrics); ok {
+		c.metrics = m
+	}
+}
+
+// Connect establishes a connection to MongoDB and registers metrics using the provided configuration when the client was Created.
+func (c *client) Connect() {
```

Comment:
This one should return an issue error instead of logging it

```suggestion
func (c *client) Connect() error {
```